src/aead/aead.rs

```diff
@@ -25,6 +25,82 @@
 //! [AEAD]: http://www-cse.ucsd.edu/~mihir/papers/oem.html
 //! [`crypto.cipher.AEAD`]: https://golang.org/pkg/crypto/cipher/#AEAD
 
+//!
+//! **Example (if you don't know what you are doing, read this first)**
+//!
+//!```rust
+//!
+//!extern crate ring;
+//!
+//!use ring::aead::*;
+//!use ring::pbkdf2::*;
+//!use ring::rand::SystemRandom;
+//!
+//!fn main() {
+//!    // The password will be used to generate a key and it is normally used when
+//!    // it is generated by the user. If you need a key which will be stored in a file you
+//!    // generate it directly.
+//!    let password = b"nice password";
+//!
+//!    // Usually the salt has some random data and something that relates to the user
+//!    // like an username and it must be 8 bytes (64 bits)
+//!    let salt = [0, 1, 2, 3, 4, 5, 6, 7];
+//!
+//!    // Keys must have 32 bytes and if you want to use an user password for it you have
+//!    // to use an algorithm to derive it (in this case pbkdf2).
+//!    let mut key = [0; 32];
+//!    derive(&HMAC_SHA256, 100, &salt, &password[..], &mut key);
+//!
+//!    // Your private data
+//!    let content = b"content to encrypt".to_vec();
+//!    println!("Content to encrypt's size {}", content.len());
+//!
+//!    // Additional data that you would like to send and it would not be encrypted but it
+//!    // would be signed
+//!    let additional_data: [u8; 0] = [];
+//!
+//!    // Ring uses the same input variable as output
+//!    let mut in_out = content.clone();
+//!
+//!    // The input/output variable need some space to store the tag which will have the
+//!    // signature
+//!    println!("Tag len {}", CHACHA20_POLY1305.tag_len());
+//!    for _ in 0..CHACHA20_POLY1305.tag_len() {
+//!        in_out.push(0);
+//!    }
+//!
+//!    // Opening key used to decrypt data
+//!    let opening_key = OpeningKey::new(&CHACHA20_POLY1305, &key).unwrap();
```

Comment:
It is counter-intuitive to initialize the `OpeningKey` here -- it should be moved behind all the encryption code. In real-world code, this would be in a different function.